modules/ting_subsearch/ting_subsearch_bibdk/ting_subsearch_bibdk.module

```diff
@@ -104,6 +102,49 @@ function ting_subsearch_bibdk_opensearch_pre_execute($request) {
   }
 }
 
+/**
+ * Implements hook_opensearch_cache_key_alter().
+ */
+function ting_subsearch_bibdk_opensearch_cache_key_alter(&$cid, $request) {
+  if (ting_subsearch_bibdk_is_bibdk_subsearch()) {
+    $cid .= ':ting_subsearch_bibdk';
+  }
+}
+
+/**
+ * Whether a bibdk subsearch is currently being performed.
+ *
+ * @param bool $new_value
+ *   Used internally to control the static variable.
+ *
+ * @return bool
+ *   TRUE if bibdk subsearch is currently being performed. FALSE otherwise.
+ */
+function ting_subsearch_bibdk_is_bibdk_subsearch($new_value = NULL) {
+  $is_bibdk_subsearch = &drupal_static(__FUNCTION__, FALSE);
+  if (isset($new_value)) {
+    $is_bibdk_subsearch = $new_value;
+  }
+  return $is_bibdk_subsearch;
+}
+
+/**
+ * Get the query of the last bibdk subsearch performed in the request.
+ *
+ * @param string $new_value
+ *   Used internally to control the static variable.
+ *
+ * @return mixed
+ *   Returns NULL if no bibdk subearch has been performed in the request.
+ */
+function ting_subsearch_bibdk_get_bibdk_subsearch_query($new_value = NULL) {
```

Comment:
I do not think it is a good idea to have a function with the verb `get` in the name which accepts a value that changes the internal state.

I suggest `ting_subsearch_bibdk_subsearch_query`.

Reply:
I agree. I wasn't really satisfied with the name either. I'll update the function name to your suggestion.